src/marten/db/query/set.cr

```diff
@@ -508,6 +508,89 @@ module Marten
           filter(query)
         end
 
+        # Returns a query set whose records match the given query node object.
+        #
+        # This method enables filtering based on raw SQL conditions, offering greater
+        # flexibility than standard field predicates. It returns a modified `Marten::DB::Query::Set`.
+        #
+        # Example:
+        #
+        # ```
+        # query_set = Post.all
+        # query_set.filter("is_published = true")
+        # ```
+        def filter(query_string : String)
```

Comment:
All these new methods should probably be added to the `Marten::DB::Model::Querying` module (so that they can be used from model classes directly). What do you think?